core/src/main/java/org/jdbi/v3/core/Jdbi.java

```diff
@@ -391,11 +413,9 @@ public Handle open() {
      * @throws X any exception thrown by the callback
      */
     public <R, X extends Exception> R withHandle(HandleCallback<R, X> callback) throws X {
+        final HandleCallback<R, X> decoratedCallback = handleCallbackDecorator.get().decorate(callback);
 
-        HandleSupplier handleSupplier = threadHandleSupplier.get();
-
-        HandleCallback<R, X> decoratedCallback = handleCallbackDecorator.get().decorate(callback);
-
+        final HandleSupplier handleSupplier = getHandleScope().get();
```

Comment:
I keep going back and forth around "should this be `HandleScope handleScope = getHandleScope()` and then use the handleScope object everywhere.

This seems to be a case of second-guessing the JVM; my hunch is that it will inline the shallow getters anyway (they are on a very hot path) so it should not actually matter..

Reply:
I am not thinking about the nano-optimization necessarily, although that's always a consideration. I am thinking along the lines of, for the resource to be managed correctly, the acquire / release calls must be paired. So it is most correct to ensure we always pair the calls to the same instance specifically, so you can store it in a local variable.